proxy/src/fully_qualified_authority.rs

```diff
@@ -15,10 +15,10 @@ impl FullyQualifiedAuthority {
     /// This assumes the authority is syntactically valid.
     pub fn new(authority: &Authority, default_namespace: Option<&str>,
```

Comment:
nit but maybe clearer to either rename or make it a module-fn?  Seems surprising for `new` to return an option.